lib/rules/selector-anb-no-unmatchable/index.mjs

```diff
@@ -85,13 +93,20 @@ const rule = (primary) => {
 							return;
 						}
 
+						const index = pseudoClassSelector.loc?.start.offset;
+						const endIndex = pseudoClassSelector.loc?.end.offset;
+
 						if (isUnmatchableNth(child.nth)) {
 							report({
 								message: messages.rejected,
-								messageArgs: [`:${pseudoClassSelector.name}`],
+								messageArgs: [
+									isNumber(index) && isNumber(endIndex)
+										? ruleSelector.slice(index, endIndex)
+										: generate(pseudoClassSelector),
+								],
 								node: ruleNode,
-								index: pseudoClassSelector.loc?.start.column,
-								endIndex: pseudoClassSelector.loc?.end.column,
```

Comment:
As far as I could tell `csstree` doesn't seem to have a way to re-serialize to source. It only has a `generate` function which takes the AST and generates equivalent CSS.

This implies that some source might alter.
For example:
```diff
- :nth-child(+0n)
+ :nth-child(0n)
```

-----

At the same time `loc` in `pseudoClassSelector.loc` seems to be possibly `undefined`.
So we can't reliably `slice` the original input using `index` and `endIndex`.

-----

I chose to navigate this by first trying to use a `slice` of the original selector when `index` and `endIndex` are numbers and not `undefined`.

If that fails we fallback to `generate()`.

-----

Alternatively we could always use `generate()` and accept that it can produce selectors in messages that are subtly different from the actual source.

This would reduce complexity a bit in this rule.

Reply:
Thanks for the elaboration. The fallback makes sense to me. 👍🏼 

Leaving a brief code comment about why this uses `generate()` seems valuable to me, but what do you think?

Reply:
After fixing https://github.com/stylelint/stylelint/pull/8003 I think I understand better why and when `loc` will be `undefined`.

If I understand it correctly it will never be `undefined` when parsing with `{ positions: true }`

So I think we can safely do this:

```js
assert(pseudoClassSelector.loc);

const index = pseudoClassSelector.loc.start.offset;
const endIndex = pseudoClassSelector.loc.end.offset;

if (isUnmatchableNth(child.nth)) {
	report({
		message: messages.rejected,
		messageArgs: [ruleSelector.slice(index, endIndex)],
		node: ruleNode,
		index,
		endIndex,
		result,
		ruleName,
	});
}
```

Then we don't have the more complex code and also don't have to explain the complexity in a comment :)

Reply:
Much better! 👍🏼